src/utilities/dynamodb-helper.js

```diff
@@ -1,13 +1,34 @@
 import { logError } from '../middleware/logging';
-import { validate } from 'uuid';
-import { getEpochTimeInSecond, getUKTimestamp } from '../services/time';
-import moment from 'moment-timezone';
-
-export const validateIds = (conversationId, messageId) => {
-  const uuidsAreValid = validate(conversationId) && validate(messageId);
-  if (!uuidsAreValid) {
-    throw new Error('received invalid uuid as either conversationId or messageId');
+import { getUKTimestamp } from '../services/time';
+import { ConversationStatus, CoreStatus, FragmentStatus, QueryKeyType } from '../constants/enums';
+import { isConversation } from '../models/conversation';
+import { isCore } from '../models/core';
+import { isFragment } from '../models/fragment';
+
+export const buildBaseQueryParams = (
+  keyValue,
+  queryKeyType = QueryKeyType.InboundConversationId
+) => {
+  const keyToken = `#${queryKeyType}`;
+  const valueToken = `:${queryKeyType}`;
+
+  const queryParams = {
+    ExpressionAttributeValues: {
+      [valueToken]: keyValue
+    },
+    ExpressionAttributeNames: {
+      [keyToken]: queryKeyType
+    },
+    KeyConditionExpression: `${keyToken} = ${valueToken}`
+  };
+
+  switch (queryKeyType) {
+    case QueryKeyType.NhsNumber:
+    case QueryKeyType.OutboundConversationId:
+      queryParams.IndexName = `${queryKeyType}SecondaryIndex`;
   }
```

Comment:
Could we add a default case if there is any chance of erroneous input of `queryKeyType`?